src/middleware.ts

```diff
@@ -259,43 +302,59 @@ export const persist = <S extends State>(
   }
 
   // rehydrate initial state with existing stored state
-  ;(async () => {
-    const postRehydrationCallback = onRehydrateStorage?.(get()) || undefined
-
-    try {
-      const storageValue = await storage.getItem(name)
 
+  // a workaround to solve the issue of not storing rehydrated state in sync storage
+  // the set(state) value would be later overridden with initial state by create()
+  // to avoid this, we merge the state from localStorage into the initial state.
+  // TODO: find a better solution for this
```

Comment:
Do we need this TODO line? It looks to me that this is a right solution.
What do you guys think?

Reply:
it's a workaround. 
Yeah, it's working. maybe we can remove `TODO`